torch/fx/passes/shape_prop.py

```diff
@@ -109,11 +109,36 @@ def forward(self, x):
 
     Args:
          module (GraphModule): The module to be executed
+         fake_mode (FakeTensorMode): A fake mode for copying the gm
 
     """
+    def __init__(self, gm, fake_mode=None):
+        super().__init__(gm)
+        if fake_mode:
+            from torch._dynamo.utils import deepcopy_to_fake_tensor
+            # Note:
+            # We need fake execution cause the inputs are fake, however, we cannot fakify the module
+            # - because we need to write to the tensor_meta of the real module. So we fakify to
+            # produce a result (L130 below), to extract tensor meta, and then keep going.
+            #
+            # If we were to fakify, we would write to the wrong node, and then downstream fusion
+            # would be missing the tensor_meta.
+            #
+            # See torch/_inductor/overrides.py for where this is called upstream of fusion.
+            self.fake_module = deepcopy_to_fake_tensor(self.module, fake_mode)
+        else:
+            self.fake_module = None
+
+        self.real_module = self.module
+
     def run_node(self, n : Node) -> Any:
         try:
+            if self.fake_module is not None:
+                # Hacky little swap lol
+                self.module = self.fake_module
+
             result = super().run_node(n)
+            self.module = self.real_module
```

Comment:
If run_node raises an exception this won't reset the module.

Reply:
There's no need to do a swap like this. The only thing you need to override is a few ops (`get_attr`, `call_module` I think?) to fetch from fake module rather than real module.

Reply:
good call

Reply:
> If run_node raises an exception this won't reset the module.

good call

> There's no need to do a swap like this. The only thing you need to override is a few ops (`get_attr`, `call_module` I think?) to fetch from fake module rather than real module.

Yeah but we already override this, this feels simpler. No strong opinion tho.